### PR TITLE
Initial API documentation.

### LINKED ISSUE #6: Document API
Document the API.

To be done on both:
- [ ]  DTO
- [ ]  Resource class

Help:
- https://dat3-2019fall.netlify.com/Flow-2/week3/03-10-2019-OpenAPI/

There should also be examples in the code. See #1

### LINKED ISSUE #7: Published the API documentation.
When #6 is done:

- Publish the documentation.

Make sure you can access the API documentation as a swagger page on the droplet. 

Maybe make a link to the documentation from the main page.

### PR DESCRIPTION
**Pull request by Runi-VN**
Fixes #6  
Fixes #7

Currently `DTO_IN` & `DTO_OUT` are equal!! really simple, have multiple constructors.  
Maybe our `DTO_OUT` should include ID for some operations, like `Add`? *(Not so important)*  
This DTO will clash with @HrBjarup & @Castau and their DTO work on #9.  

Most CRUD-operations expect a JSON body, if we want simple fields instead (e.g. FindHobbyByID) we can create that too.

### Errors:  
- All DTOs *List*'s are currently set to the following:  
```@Schema(example="None available")```  
  
I would like some help to JSONIFY some examples. 
- The `Person`-Schema does not list the DTO requirements or examples???

### CODE REVIEW

File: src/main/java/dto/MovieInfo.java [hunk 1]
```diff
@@ -9,6 +10,8 @@
 */
 
 @Schema(name = "MovieInfo")  //Because of this we could have called the class MovieInfoDTO
+@io.swagger.v3.oas.annotations.Hidden
+@Deprecated
```
Comment (MalteMagnussen) [inline src/main/java/dto/MovieInfo.java]: Smart

File: src/main/java/dto/HobbyDTO_IN.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.List;
+import java.util.Objects;
+
+@Schema(name = "Hobby")
+public class HobbyDTO_IN {
+    
+    @Schema(required = true,example = "Lacrosse")
+    private String name;
+    @Schema(required = true,example = "Rich People Game")
+    private String description;
```
Comment (MalteMagnussen) [inline src/main/java/dto/HobbyDTO_IN.java]: Hilarious

File: src/main/java/rest/SearchResource.java [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                title = "CA2 API",
+                version = "1.0",
+                description = "API related to Cphbusiness 3rd semester CS assignment 'CA2'.",
+                contact = @Contact(name = "Github Contributors", url = "https://github.com/MalteMagnussen/CA2/")
+        ),
+        tags = {
+            @Tag(name = "General", description = "API related to CA2"),
+            @Tag(name = "Persons", description = "CRUD-operations for Person"),
+            @Tag(name = "Hobbies", description = "CRUD-operations for Hobby"),
+            @Tag(name = "Movies", description = "Deprecated")
```
Comment (MalteMagnussen) [inline src/main/java/rest/SearchResource.java]: Nice. Eventually we'll remove it. Just wanted it in here in the beginning as an example and a jump-off point.

File: src/main/java/rest/SearchResource.java [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+        },
+        servers = {
+            @Server(
+                    description = "For Local host testing",
+                    url = "http://localhost:8080/startcodeoas"
+            ),
+            @Server(
+                    description = "Server API",
+                    url = "https://maltemagnussen.com/CA2/"
```
Comment (MalteMagnussen) [inline src/main/java/rest/SearchResource.java]: Shouldn't this URL be something like: https://maltemagnussen.com/CA2/api/search

File: src/main/java/rest/SearchResource.java [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+                @ApiResponse(responseCode = "400", description = "Not all arguments provided with the body")
+            })
+    public MovieInfo createMovie(MovieInfo movie) {
+        if (movie.getTitle() == null || movie.getDirector() == null || movie.getYear() == null) {
+            throw new WebApplicationException("Not all required arguments included", 400);
+        }
+        movie.setId(464);
+        return movie;
+    }
+//</editor-fold>
```
Comment (MalteMagnussen) [inline src/main/java/rest/SearchResource.java]: To be removed before last release.

File: src/main/java/rest/SearchResource.java (hunk 4 above)
Comment (Runi-VN) [reply src/main/java/rest/SearchResource.java]: idk tbh

**Pull request merged**